File: src/main/java/com/hepdd/gtmthings/api/machine/IWirelessEnergyContainerHolder.java
package com.hepdd.gtmthings.api.machine;

import com.hepdd.gtmthings.api.misc.WirelessEnergyContainer;

import java.util.UUID;

public interface IWirelessEnergyContainerHolder {

    UUID getUUID();

    void setWirelessEnergyContainerCache(WirelessEnergyContainer container);

    WirelessEnergyContainer getWirelessEnergyContainerCache();

    default WirelessEnergyContainer getWirelessEnergyContainer() {
        if (getWirelessEnergyContainerCache() == null) {
            WirelessEnergyContainer container = WirelessEnergyContainer.getOrCreateContainer(getUUID());
            setWirelessEnergyContainerCache(container);
        }
        return getWirelessEnergyContainerCache();
    }
}

File: src/main/java/com/hepdd/gtmthings/api/machine/IWirelessEnergyContainerHolder.kt
package com.hepdd.gtmthings.api.machine;

import com.hepdd.gtmthings.api.capability.IBindable;
import com.hepdd.gtmthings.api.misc.WirelessEnergyContainer;

import javax.annotation.Nullable;

public interface IWirelessEnergyContainerHolder extends IBindable {

    void setWirelessEnergyContainerCache(WirelessEnergyContainer container);

    WirelessEnergyContainer getWirelessEnergyContainerCache();

    @Nullable
    default WirelessEnergyContainer getWirelessEnergyContainer() {
        if (getUUID() != null && getWirelessEnergyContainerCache() == null) {
            WirelessEnergyContainer container = WirelessEnergyContainer.getOrCreateContainer(getUUID());
            setWirelessEnergyContainerCache(container);
        }
        return getWirelessEnergyContainerCache();
    }
}
